rpcs3/Emu/Io/usb_device.h

```diff
@@ -153,16 +154,17 @@ struct UsbDescriptorNode
 		}
 		return nodesize;
 	}
-	void write_data(u8*& ptr)
+
+	u32 write_data(u8* ptr, u32 max_size) const
 	{
-		ptr[0] = bLength;
-		ptr[1] = bDescriptorType;
-		memcpy(ptr + 2, data, bLength - 2);
-		ptr += bLength;
-		for (auto& node : subnodes)
+		u32 size = std::min<u32>(bLength, max_size);
+		memcpy(ptr, this, size);
+		for (const auto& node : subnodes)
 		{
-			node.write_data(ptr);
+			if (max_size == 0) break;
```

Comment:
```cpp
const u32 size_left = max_size - size;
if (size_left == 0) break;
size += node.write_data(ptr + size, size_left);
```

Reply:
They seem to mean the same thing (`size_left` vs `max_size - size`).

Reply:
yes but you're checking max_size in current version, not size_left, and since you use max_size-size twice might as well put it in a variable for clarity.

Reply:
Right. My mistake.
Thank you for pointing it out!